containers/python/python_container.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import print_function
+import rpc
+import os
+import sys
+
+import numpy as np
+np.set_printoptions(threshold=np.nan)
+
+sys.path.append(os.path.abspath("/lib/"))
+import pywrencloudpickle
+
+
+def load_predict_func(file_path):
+    with open(file_path, 'r') as serialized_func_file:
+        return pywrencloudpickle.load(serialized_func_file)
+
+
+class PredictContainer(rpc.ModelContainerBase):
+    def __init__(self, path, input_type):
+        print("Initializing prediction container")
+
+        self.input_type = rpc.string_to_input_type(input_type)
+
+        predict_fname = "predict.txt"
+        predict_path = "{dir}/{predict_fname}".format(
+            dir=path, predict_fname=predict_fname)
+
+        self.predict_func = load_predict_func(predict_path)
+
+        print("Loaded prediction function")
+
+    def predict_ints(self, inputs):
+        if self.input_type != rpc.INPUT_TYPE_INTS:
+            pass
+        return np.asarray(self.predict_func(inputs)).astype(np.float32)
+
+    def predict_floats(self, inputs):
+        if self.input_type != rpc.INPUT_TYPE_FLOATS:
+            pass
+        return np.asarray(self.predict_func(inputs)).astype(np.float32)
+
+    def predict_doubles(self, inputs):
+        if self.input_type != rpc.INPUT_TYPE_DOUBLES:
+            pass
+        return np.asarray(self.predict_func(inputs)).astype(np.float32)
+
+    def predict_bytes(self, inputs):
+        if self.input_type != rpc.INPUT_TYPE_BYTES:
+            pass
+        return np.asarray(self.predict_func(inputs)).astype(np.float32)
+
+    def predict_strings(self, inputs):
+        if self.input_type != rpc.INPUT_TYPE_STRINGS:
+            pass
+        return np.asarray(self.predict_func(inputs)).astype(np.float32)
+
+
+if __name__ == "__main__":
+    print("Starting PredictContainer container")
+    try:
+        model_name = os.environ["CLIPPER_MODEL_NAME"]
+    except KeyError:
+        print(
+            "ERROR: CLIPPER_MODEL_NAME environment variable must be set",
+            file=sys.stdout)
+        sys.exit(1)
+    try:
+        model_version = os.environ["CLIPPER_MODEL_VERSION"]
+    except KeyError:
+        print(
+            "ERROR: CLIPPER_MODEL_VERSION environment variable must be set",
+            file=sys.stdout)
+        sys.exit(1)
+
+    ip = "127.0.0.1"
+    if "CLIPPER_IP" in os.environ:
+        ip = os.environ["CLIPPER_IP"]
+    else:
+        print("Connecting to Clipper on localhost")
+
+    port = 7000
+    if "CLIPPER_PORT" in os.environ:
+        port = int(os.environ["CLIPPER_PORT"])
+    else:
+        print("Connecting to Clipper with default port: 7000")
```

Comment:
Use string formatting and print the `port` variable instead of hardcoding 7000 into the string